### ISSUE TITLE
Disable dexcount for certain builds

### ISSUE THREAD

**Issue by rafagcamargo** [KeepSafe/dexcount-gradle-plugin#235]
Hi, this is more a question than a issue itself. I couldn't find a way to disable dexcount for a specific build type or flavor. Is it possible?

Thank you and sorry for open an issue to ask this question.

**Comment by benjamin-bader** [comment]
You have two options currently, neither of them amazing:

1. Apply the dexcount plugin conditionally in your gradle file, e.g. in app/build.gradle:
```
if (shouldCountMethods()) {
  apply plugin: 'com.getkeepsafe.dexcount'
}
```

2. Just suppress the dexcount task on the command line: `./gradlew assembleRelease -x countReleaseDexMethods`.

**Comment by jaredsburrows** [comment]
I agree with @benjamin-bader. This can be easily handled on the developers end.

**Comment by rafagcamargo** [comment]
Thank you! I endedup creating a gradle external property in order to disable for debug builds and enable only for CI builds. Have you guys considered a plug-in property to disable count on debug?

**Comment by benjamin-bader** [comment]
I'd be happy to take a PR implementing such a feature!  Can I ask what your use-case is for disabling counts on debug?

**Comment by rafagcamargo** [comment]
Sure, I have splitApk enabled for 3 architectures and my local build was taking 2:30 minutes... decided to profile it and ~ 50 seconds was dexCount running for each APK, three in the case. I would to like to remove this burden from local builds.

**Comment by benjamin-bader** [comment]
I'm surprised to hear that counts for three APKs took so long!  That is so slow, I'd consider it to be a bug.  Do I understand correctly, that there are only three APKs being counted?  Are you multidexing?  If so, how many methods are involved here?  Thanks!

**Comment by ankushg** [comment]
I would also appreciate some configuration options to enable/disable the task.

I only really care about significant changes to these metrics for proguarded release builds, and would rather save the 15 seconds of debug build time.

Currently, I'm skipping the plugin application for debug builds as follows:

```groovy
if (!getGradle().getStartParameter().getTaskRequests().toString().toLowerCase().contains("debug")) {
    apply plugin: 'com.getkeepsafe.dexcount' // Must come after Android plugin, don't apply to debug builds
}
```

**Comment by ankushg** [comment]
Would it be reasonable to check something like `ext.enableDexcount` in the `addDexcountTaskToGraph` method (defaulting it to `true` so we don't break backwards compatibility)? 

That would allow developers to set it to `false` for time-sensitive variants.

**Comment by jaredsburrows** [comment]
I still believe Gradle properties work best for this. For this plugin, it is meant to count and display the number of methods. I do not believe we should add specific flags for builds. Gradle properties were designed to handle external changes and flags: https://docs.gradle.org/current/userguide/build_environment.html

**Comment by benjamin-bader** [comment]
Hi folks, in PR #255 we'll be adding a configuration option for this.  It should be available in dexcount 0.8.6, to be released (hopefully) tomorrow.
